lib/server.cpp

```
@@ -0,0 +1,286 @@
+#include <iostream>
+#include <stdlib.h>
+#include <sys/socket.h>
+#include <arpa/inet.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <netinet/in.h>
+#include <unistd.h>
+#include <cstring>
+#include <regex>
+#include <fstream>
+
+#include <openssl/ssl.h>
+#include <openssl/err.h>
+
+#include "server.h"
+
+using std::cout;
+using std::string;
+using std::regex;
+using std::smatch;
+using std::stoi;
+using Socialite::NetStream;
+
+struct connection_thread_h {
+	Socialite::Server* server;
+	NetStream* stream;
+};
+
+// implied SSL given cert and key file paths
+Socialite::Server::Server(int portNumber, string cert, string key, int maxThreads, int flags) {
+	streamPolicy = std::pair<string, int>("none", 0);
+	threads = new Juggler::ThreadManager(maxThreads + 1); // +1 for listening thread
+	mainLoopThreadId = -1;
+	setupError = false;
+	if(flags & S_ONE_THREAD_ONLY) {
+		threadPerConnection = true;
+	}
+	else {
+		threadPerConnection = false;
+	}
+	port = portNumber;
+	runningSSL = true;
+	waitingCount = 4;
+
+	if(!setupSSL(cert, key)) {
+		setupError = true;
+		printf("ERROR: setupSSL\n");
```

It'd be great to `throw` and error rather than just printing it